subprojects/internal-integ-testing/src/test/groovy/org/gradle/integtests/fixtures/extensions/MultiTestLifecycleSpec.groovy

```diff
@@ -59,16 +68,40 @@ class MultiTestLifecycleSpec extends Specification {
         true
     }
 
-    def unrolled() {
+    def "unrolled foo: #foo"() {
         LIFECYCLE.pushEvent("unrolled test: $foo: isFluid: ${FluidDependenciesResolveInterceptor.isFluid()}")
         expect:
         true
 
         where:
-        foo << [1, 2]
+        foo << [1, 2, 3]
+    }
+
+    @RequiredFeature(feature = FluidDependenciesResolveInterceptor.ASSUME_FLUID_DEPENDENCIES, value = "true")
+    def "unrolled foo with required: #foo"() {
+        LIFECYCLE.pushEvent("unrolled test with required: $foo: isFluid: ${FluidDependenciesResolveInterceptor.isFluid()}")
+        expect:
+        true
+
+        where:
+        foo << [1, 2, 3]
+    }
+
+    def "skipped test"() {
+        LIFECYCLE.pushEvent("skipped test: isFluid: ${FluidDependenciesResolveInterceptor.isFluid()}")
+        Assume.assumeTrue("can skip test", false)
+
+        expect:
+        false
     }
 
     static class SampleRule extends ExternalResource {
+        private final String name
```

Comment:
Is this necessary? I guess the iteration name provider doesn't work without this?

Reply:
Everything works fine without this, though the test didn't catch a problem I introduced in an intermediate step which caused Spock to not run the initialization of the class. Then JUnit called the default constructor on this rule, hiding the problem. I added a clarifying comment.